src/fieldExtensions.js

```diff
@@ -0,0 +1,200 @@
+ *   - actionType: { ...state, actionType: 'LAST_ACTION' }
+ *   - action: { ...state, action: { type: 'LAST_ACTION', ...actionPayload } }
+ *   - inline: { ...state, present: { action: { type: 'LAST', ...payload }, ...otherFields } }
+ *
+ * @param {actionFieldIncludeAction} config.includeAction - A filter function that decides if
+ * the action is inserted into history.
+ */
+export const actionField = ({ insertMethod, includeAction } = {}) => {
+  if (insertMethod === 'inline') {
+    return inlineActionField({ includeAction })
+  }
+
+  let extend
+  if (insertMethod === 'action') {
+    extend = (state, action) => ({ ...state, action })
+  } else if (!insertMethod || insertMethod === 'actionType') {
+    extend = (state, action) => ({ ...state, actionType: action.type })
+  } else {
+    throw new Error(
+      `Unrecognized \`insertMethod\` option for actionField() extension: ${insertMethod}.\n` +
+      'Options are "action", "inline", or "actionType"'
+    )
+  }
+
+  return (undoableConfig) => {
+    const included = includeAction || (() => true)
+
+    if (!undoableConfig.disableWarnings) {
+      console.warn(
+        'Warning: the actionField() extension might override other state fields',
+        'such as "action", "present.action", or "actionType".\n',
+        'Disable this warning by passing `disableWarnings: true` into the undoable config'
+      )
+    }
+
+    let lastAction = {}
+
+    return (state, action) => {
+      if (included(action)) {
+        lastAction = action
+        return extend(state, action)
+      }
+
+      return extend(state, lastAction)
+    }
+  }
+}
+
+/**
+ * @private
+ */
+const inlineActionField = ({ includeAction } = {}) => {
+  return (undoableConfig) => {
+    if (!undoableConfig.disableWarnings) {
+      console.warn(
+        'Warning: the actionField() extension might override other state fields',
+        'such as "action", "present.action", or "actionType".\n',
+        'Disable this warning by passing `disableWarnings: true` into the undoable config'
+      )
+    }
+
+    const ignoredActions = [
+      undoableConfig.undoType,
+      undoableConfig.redoType,
+      undoableConfig.jumpType,
+      undoableConfig.jumpToPastType,
+      undoableConfig.jumpToFutureType,
+      ...undoableConfig.clearHistoryType,
+      ...undoableConfig.initTypes
+    ]
+
+    const included = includeAction || (() => true)
+
+    return (state, action) => {
+      if (included(action) && ignoredActions.indexOf(action.type) === -1) {
+        const newState = { ...state, present: { ...state.present, action } }
+
+        if (state.present === state._latestUnfiltered) {
+          newState._latestUnfiltered = newState.present
+        }
+        return newState
+      }
+
+      return state
+    }
+  }
+}
+
+// istanbul ignore next: This will be put on hold for now...
+// eslint-disable-next-line no-unused-vars
+const nullifyFields = (fields = [], nullValue = null) => {
```

Comment:
document these in README as well?

do you think the field extensions should be part of redux-undo core, or maybe a separate library? I don't think everyone will need it and it slightly increases the bundle size

Reply:
I had briefly mentioned in the PR description that these were WIP and would come out soon, but I realize that it is silly to leave them there. I'll remove them until they are fully implemented. :+1: 

As for separating them into a different package, that sounds like a good idea! They are purposely optional and, yes, not everyone needs it in their bundle (though tree-shaking is starting to mitigate the problem).